addons/website_sale/models/product_template.py

```diff
@@ -194,7 +194,7 @@ def _get_sales_prices(self, pricelist):
                 'price_reduce': price_reduce
             }
             base_price = None
-            price_list_contains_template = price_reduce != base_sales_prices[template.id]
+            price_list_contains_template = price_reduce != taxes.compute_all(base_sales_prices[template.id], pricelist.currency_id, 1, template, partner_sudo)[tax_display]
```

Comment:
Instead of applying the taxes on the amount, i suppose this code could simply moved before the taxes application on `price_reduce`:

```python
price_reduce = sales_prices[template.id]
price_list_contains_template = price_reduce != base_sales_prices[template.id]
```

Also, could you include a little test to cover this bugfix please?